dot-net/Centroid/Config.cs

```diff
@@ -78,7 +78,23 @@ public override bool TryConvert(ConvertBinder binder, out object result)
 
             try
             {
-                result = Convert.ChangeType(RawConfig, binder.Type);
+                JToken rawConfig = RawConfig;
+                result = rawConfig.ToObject(binder.Type);
+                //ensure at least one property in the config bound to the type
+                var JResult = JToken.FromObject(result);
```

Comment:
What do you think of rolling these casts up into .Cast<JProperty> calls like this instead:

```cs
// Ensure at least one property in the config bound to the type
var jResult = JToken.FromObject(result);
var hasBoundProperty = jResult.Cast<JProperty>().Any(property => {
        return rawConfig.Cast<JProperty>().Any(rawProperty => {
            return NormaliseKey(rawProperty.Name).Equals(NormaliseKey(property.Name)) && rawProperty.Value.Equals(property.Value);
        });
    });
```